src/save/serializeNmriumState.ts

```diff
@@ -0,0 +1,132 @@
+import {
+  NmriumData,
+  NmriumDataSource,
+  NmriumRawData,
+  NmriumState,
+  Spectra,
+  SpectraRawData,
+  SpectraSource,
+} from '../types/NmriumState';
+import {
+  Spectrum1DRawData,
+  Spectrum1DSource,
+} from '../types/Spectra/Spectrum1D';
+import {
+  Spectrum2DRawData,
+  Spectrum2DSource,
+} from '../types/Spectra/Spectrum2D';
+import { getBaseFile } from '../utilities/getBaseFile';
+
+const NO_DATA = 'noData';
+const RAW_DATA = 'rawData';
+const DATA_SOURCE = 'dataSource';
+
+export type IncludeData = 'rawData' | 'dataSource' | 'noData';
+
+const defaultContent = [
+  'id',
+  'display',
+  'filters',
+  'ranges',
+  'integrals',
+  'peaks',
+  'correlations',
+];
+
+export interface GetNmriumStateOptions {
+  includeData?: IncludeData;
+  includeSettings?: boolean;
+  includeView?: boolean;
+}
+
+export function serializeNmriumState(
+  nmriumState: NmriumState,
+  options: GetNmriumStateOptions,
+) {
+  const { version, data } = nmriumState;
+
+  const newNmriumState: NmriumState = {
+    version,
+  };
+
+  const { includeData } = options;
+  if (data && includeData !== NO_DATA) {
+    newNmriumState.data = getNmriumData(data, includeData);
+  }
+  const { includeSettings, includeView } = options;
+
+  if (includeSettings) newNmriumState.settings = nmriumState.settings;
+  if (includeView) newNmriumState.view = nmriumState.view;
+
+  return newNmriumState;
+}
+
+function getNmriumData(
+  nmriumData: NmriumData | NmriumDataSource | NmriumRawData,
+  includeData?: IncludeData,
+) {
+  const { spectra, molecules } = nmriumData;
+
+  switch (includeData) {
+    case DATA_SOURCE:
+      return {
+        molecules,
+        spectra: spectra.map(getDataSource),
+      } as NmriumDataSource;
+    case RAW_DATA:
+      return {
+        molecules,
+        spectra: spectra.map(getRawData),
+      } as NmriumRawData;
+    default:
+      return {
+        molecules,
+        spectra,
+      } as NmriumData;
+  }
+}
+
+function getDataSource(spectrum: Spectra | SpectraSource | SpectraRawData) {
+  if (!spectrum.source) {
+    throw new Error('to export data source, it should exists');
+  }
+  if (!spectrum.source?.baseURL) {
+    throw new Error('to export data source baseURL should exists');
+  }
+  const result = maybeAdd({}, spectrum, defaultContent) as
+    | Spectrum1DSource
+    | Spectrum2DSource;
+
+  const { files } = spectrum.source;
+  result.source = {
+    ...spectrum.source,
+    files: files.map(getBaseFile),
+  };
+  return result;
+}
+
+function getRawData(nmriumData: Spectra | SpectraSource | SpectraRawData) {
+  const { data, info, originalData, originalInfo } = nmriumData;
+  return maybeAdd(
+    {
+      data: originalData || data,
+      info: originalInfo || info,
+    },
+    nmriumData,
+    defaultContent.concat(['meta', 'metaInfo']),
+  ) as Spectrum1DRawData | Spectrum2DRawData;
+}
+
+function maybeAdd(
+  result: Record<string, any>,
+  source: Record<string, any>,
+  list = defaultContent,
```

Comment:
```suggestion
  propertiesToKeep,
```

The default is unused, and I think it's more clear to always pass the list.